source/matrix_free/vector_data_exchange.cc

```diff
@@ -316,6 +318,965 @@ namespace internal
             ghost_array[j] = 0.;
       }
 
+
+
+      namespace internal
+      {
+        void
+        compress(std::vector<unsigned int> &recv_sm_ptr,
```

Comment:
Can you please adapt the name of this function? I think a better name could be `compress_to_contiguous_ranges`.